src/nvim/main.c

```diff
@@ -270,6 +270,13 @@ int main(int argc, char **argv)
 
   if (embedded_mode) {
     const char *err;
+    if (exmode_active) {
+      // exmode_active being different from 0 here means that -e/-E (read from
+      // stdin) flags are present
+      mch_errmsg(_("Error: flags -e, -E, -es and -Es are incompatible with "
+            "--embed.\n"));
+      mch_exit(2);
+    }
```

Comment:
`silent_mode`  signals `-es`/`-Es`.

Could be done in `command_line_scan()`, to avoid some noise in `main()`.

Reply:
In my opinion it's better to keep the code that parses command line arguments and the code that acts on the parsed representation separate. But I don't care enough and I'm not the one who has to maintain neovim (for now :p) so I won't argue :).

Reply:
maybe, but command_line_scan() has numerous such cases already--so I was thinking it's more consistent to do it in there.